pkg/kube/kclient/index_test.go

```diff
@@ -168,7 +168,7 @@ func TestIndexFilters(t *testing.T) {
 	})
 	pc := clienttest.NewWriter[*corev1.Pod](t, c)
 	c.RunAndWait(test.NewStop(t))
-	index := kclient.CreateStringIndex[*corev1.Pod](pods, func(pod *corev1.Pod) []string {
+	index := kclient.CreateStringIndex[*corev1.Pod](pods, "podIp", func(pod *corev1.Pod) []string {
```

Comment:
For these common indexes on informers, we may want to make a common package of 'wellknown' indexes. This way its easy to know when we can share an index and its much harder to accidentally define the same name but different extraction

Reply:
It seems like a good idea, but would probably be more useful for  `krt`? There seem to be very few indexes we use on `kclient` (only 4 indexes not in tests), this one is actually only used in tests. With extended `krt` usage we would probably reduce the amount of `kclient` indexes we create.